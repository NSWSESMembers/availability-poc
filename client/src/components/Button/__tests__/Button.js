import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Button
      text="text"
      onPress={noop}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <Button
      type="secondary"
      text="text"
      onPress={noop}
      disabled
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
