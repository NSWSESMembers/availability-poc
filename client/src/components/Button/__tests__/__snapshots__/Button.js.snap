// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f48603",
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    text
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1c87c9",
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    text
  </Text>
</View>
`;
